rush-plugins/rush-serve-plugin/src/phasedCommandHandler.ts

```diff
@@ -0,0 +1,134 @@
+      if (!context.isInitial || !context.isWatch) {
+        return operations;
+      }
+
+      const certManager: CertificateManager = new CertificateManager();
+
+      const certificate: ICertificate = await certManager.ensureCertificateAsync(
+        true, // Allow generating a new certificate
+        logger.terminal
+      );
+
+      const { customParameters } = context;
+
+      let requestedPort: number = 0;
+
+      const portParameter: CommandLineStringParameter | undefined = portParameterLongName
+        ? (customParameters.get(portParameterLongName) as CommandLineStringParameter)
+        : undefined;
+      if (portParameter) {
+        const rawValue: string | undefined = portParameter.value;
+        const parsedValue: number = rawValue ? parseInt(rawValue, 10) : 0;
+        if (!isNaN(parsedValue)) {
+          requestedPort = parsedValue;
+        }
+      }
+
+      // TODO: Update this to read from a custom parameter with parameterLongName matching portParameterLongName,
+      // if it exists.
+      // Requires the ability to access custom parameters from IPhasedCommand
+
+      const app: express.Express = express();
+
+      // Configure server based on project-level configs
+      // TODO: Update this to be the set of projects selected for the command
+      // Requires a hook to which the project selection is a parameter, minimally
+      const selectedProjects: ReadonlySet<RushConfigurationProject> = new Set(rushConfiguration.projects);
+
+      const serveConfig: RushServeConfiguration = new RushServeConfiguration();
+
+      const routingRules: Iterable<IRoutingRule> = await serveConfig.loadProjectConfigsAsync(
+        selectedProjects,
+        logger.terminal
+      );
+
+      function setHeaders(response: express.Response, path: string, stat: unknown): void {
+        response.set('Access-Control-Allow-Origin', '*');
+        response.set('Access-Control-Allow-Methods', 'GET');
+      }
+
+      for (const rule of routingRules) {
+        app.use(
+          rule.servePath,
+          express.static(rule.diskPath, {
+            dotfiles: 'ignore',
+            immutable: rule.immutable,
+            index: false,
+            redirect: false,
+            setHeaders
+          })
+        );
+      }
+
+      const server: https.Server = https.createServer(
+        {
+          cert: certificate.pemCertificate,
+          key: certificate.pemKey
+        },
+        app
+      );
+
+      server.listen(requestedPort);
+      await once(server, 'listening');
+
+      const address: AddressInfo | undefined = server.address() as AddressInfo;
+      activePort = address?.port;
+
+      logger.terminal.writeLine(`Content is being served from:\n  https://localhost:${activePort}/`);
+      if (portParameter) {
+        // Hack the value of the parsed command line parameter to the actual runtime value.
+        (portParameter as unknown as { _value?: string })._value = `${activePort}`;
```

Comment:
I don't like having to do this, but the CLI parameter contracts only define a getter for the `value` property, I guess under the assumption that there isn't a use case for mutating the values after parse time. However, there is utility to it here on account of the values being read back out and forwarded.

Reply:
`ts-command-line` lets you specify a default value. Would that work?

Or maybe we should just expose a function for updating/overriding the parameter value.

Reply:
This logic executes after the initial parse (since I read the value from the parameter at first), but before the parameter values get forwarded to the underlying operation runners. Would probably need to expose the function for changing the parameter value.